### PR TITLE
Fix kubeadm swallowed errors

### PR DESCRIPTION
**Pull request by alrs**
**What this PR does / why we need it**: Fixes nine swallowed errors in kubeadm, and adds descriptive error returns to Init.Run().

**Special notes for your reviewer**: I've resubmitted this PR after it required a rebase. Previously, I submitted this PR as https://github.com/kubernetes/kubernetes/pull/52591

```release-note NONE
```

### CODE REVIEW

**Comment by errordeveloper** [comment]
cc @luxas

**Comment by alrs** [comment]
@errordeveloper, @luxas: Does this PR look merge-ready to you?

**Review by luxas — COMMENTED** [review]
Thanks for the ping, now noticed this PR among my ever-growing backlog of unread notifications.

Left a couple of comments, please fix.
Thank you for this contribution :smile:!

**Comment by luxas** [comment]
(also, please squash all commits into one)

File: cmd/kubeadm/app/phases/upgrade/staticpods.go [hunk 1]
```diff
@@ -124,7 +124,9 @@ func StaticPodControlPlane(waiter apiclient.Waiter, pathMgr StaticPodPathManager
 	// Write the updated static Pod manifests into the temporary directory
 	fmt.Printf("[upgrade/staticpods] Writing upgraded Static Pod manifests to %q\n", pathMgr.TempManifestDir())
 	err = controlplane.CreateInitStaticPodManifestFiles(pathMgr.TempManifestDir(), cfg)
-
+	if err != nil {
+		return err
```
Comment (timothysc) [inline cmd/kubeadm/app/phases/upgrade/staticpods.go]: Why did you not re-wrap here?

**Review by timothysc — CHANGES_REQUESTED** [review]
Please squash commits and address minor nits.

File: cmd/kubeadm/app/phases/upgrade/staticpods.go (hunk 1 above)
Comment (alrs) [reply cmd/kubeadm/app/phases/upgrade/staticpods.go]: @timothysc got it, thanks.

File: cmd/kubeadm/app/cmd/init.go [hunk 2]
```diff
@@ -283,7 +283,7 @@ func (i *Init) Run(out io.Writer) error {
 	realCertsDir := i.cfg.CertificatesDir
 	certsDirToWriteTo, kubeConfigDir, manifestDir, err := getDirectoriesToUse(i.dryRun, i.cfg.CertificatesDir)
 	if err != nil {
-		return err
+		return fmt.Errorf("error getting directories to use:%v", err)
```
[the remark this answers is not in the document]
Comment (alrs) [reply cmd/kubeadm/app/cmd/init.go]: I was mimicking the style found elsewhere in this file, I'll change it everywhere.

**Comment by alrs** [comment]
@timothysc I think I've got this PR where you want it. The Github UI shows that "Required Github CI test is not green: pull-kubernetes-bazel-test", though I see that "pull-kubernetes-bazel-test" has a green checkmark, so 🤷

**Review by luxas — APPROVED** [review]
/lgtm

**Pull request merged**